app/components/ui/buttons.tsx

```diff
@@ -0,0 +1,44 @@
+import type { SerializeFrom } from '@remix-run/node';
+import { useRouteLoaderData } from '@remix-run/react';
+import { useEffect } from 'react';
+
+import type { loader } from '~/root';
+
+export function SignInWithGoogleButton() {
```

Comment:
Do we want to move this button into the `modules/session` folder? It's definitely a re-usable button, but I feel it's closely tied together with the session code!

Reply:
Makes sense to me! I've moved the button to be under `modules/session` instead!